internal/controller/mcpserver_controller.go

```diff
@@ -64,8 +67,16 @@ func (r *MCPServerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		}
 		logger.Error(err, "unable to fetch MCPServer")
 		return ctrl.Result{}, err
+
 	}
 
+	defer func() {
+		logger.V(1).Info("Attempting to update MCPServer status with conditions", "conditions", mcpServer.Status.Conditions)
+		if err = r.Status().Update(ctx, mcpServer); err != nil {
+			logger.Error(err, "unable to update status")
+		}
+	}()
+
```

Comment:
I wouldn't use `defer` here because if there is an error while updating the status, the code will swallow the error which prevents the controller from reacting correctly to the failure. I would instead move the `r.Status().Update()` call to the end of the function.

It's also a good idea to make a copy of the original status so we can compare the new status against it later and only update the MCPServer resource if its status has actually changed. 

```suggestion
	// Make a copy of the status to compare against later
	originalStatus := mcpServer.Status.DeepCopy()
```